### PR TITLE
updating mailer reponse tag

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

### Added
- Mailer now supports custom response tag `mailerSuccessRedirectUrl`.

### Fixed
- Forms JS breaking on the frontend when API response returns JSON but the response is not an object.
- SCSS linter errors.

### Updated
- `@infinum/eightshift-frontend-libs` to the latest version.
- `@infinum/eightshift-forms-utils` to the latest version.
- `husky` to the latest version.
- `webpack` to the latest version.
- `webpack-cli` to the latest version.
- `reactflow` to the latest version.

## Summary by CodeRabbit

- **New Features**
	- Added support for custom response tag `mailerSuccessRedirectUrl` in Mailer.
- **Bug Fixes**
	- Fixed issue with Forms JS breaking on non-object JSON responses.
	- Addressed SCSS linter errors across various components.
- **Refactor**
	- Updated response handling logic in multiple admin assets to improve JSON response processing.
	- Simplified CSS by consolidating margin and padding properties in several components.
- **Chores**
	- Updated various dependencies to their latest versions.
	- Updated `minimum_supported_wp_version` to `6.0`.
- **Documentation**
	- Minor version increment from `3.0.0` to `3.0.1` documented.

### CODE REVIEW

File: src/Blocks/components/form/assets/utils.js [hunk 1]
```diff
@@ -778,6 +778,52 @@ export class Utils {
...
+	 *
+	 * @throws Error.
+	 *
+	 * @returns {void}
+	 */
+	formSubmitIsJsonString(response, type, formId) {
+		try {
+			return JSON.parse(response);
+		} catch (e) {
+			if (formId !== null) {
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/form/assets/utils.js:798]: Please invert this `if`

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]

File: src/Blocks/components/form/assets/utils.js (hunk 1 above)
Comment (iruzevic) [reply src/Blocks/components/form/assets/utils.js:798]: I don' understand?

Comment (kancijan) [reply src/Blocks/components/form/assets/utils.js:798]:
```suggestion
			if (formId === null) {
				throw new Error(`API response returned JSON but it was malformed for this request. Function used: "${type}"`);
			}
```

**Review by kancijan — CHANGES_REQUESTED** [review]

File: src/Blocks/components/form/assets/utils.js (hunk 1 above)
Comment (iruzevic) [reply src/Blocks/components/form/assets/utils.js:798]: well this is not correct. Error will be thrown no matter if you have formId or not. 

Usage of this is that if formId is used we need to remove loader, global msg etc and throw error if this is a global form we only need to throw and error. So you suggestions are not correct

**Review by kancijan — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,24 @@ All notable changes to this project will be documented in this file.
 
 This projects adheres to [Semantic Versioning](https://semver.org/) and [Keep a CHANGELOG](https://keepachangelog.com/).
 
+## [3.0.1]
+
+### Added
+- Mailer now supports custom response tag `mailerSuccessRedirectUrl`.
+- Locations used url to settings page and top admin bar.
```
Comment (piqusy) [inline CHANGELOG.md:11]:
```suggestion
- Locations used URL to settings page and top admin bar.
```

File: src/Blocks/components/form/assets/utils.js [hunk 3]
```diff
@@ -778,6 +778,52 @@ export class Utils {
...
+				this.hideLoader(formId);
+
+				// Set global msg.
+				this.setGlobalMsg(
+					formId,
+					this.state.getStateSettingsFormServerErrorMsg(),
+					'error'
+				);
+
+				// Reset timeout for after each submit.
```
Comment (piqusy) [inline src/Blocks/components/form/assets/utils.js:812]:
```suggestion
				// Reset timeout after each submit.
```

File: src/Rest/Routes/Integrations/Mailer/FormSubmitMailer.php [hunk 4]
```diff
@@ -191,4 +193,56 @@ private function sendConfirmationEmail(string $formId, array $params, array $fil
...
+			if ($successRedirect) {
+				$successRedirectUrl = \add_query_arg(
+					[
+						'es-data' => $successRedirect,
+					],
+					$successRedirectUrl
+				);
+			}
+
+			// Add variation data, this filter will not take in effect if the success redirect variation set in the block editor.
```
Comment (piqusy) [inline src/Rest/Routes/Integrations/Mailer/FormSubmitMailer.php:231]:
```suggestion
			// Add variation data, this filter will not take in effect if the success redirect variation isn't set in the block editor.
```

**Review by piqusy — CHANGES_REQUESTED** [review]
Nice work 👏🏼 

Left a few wording suggestions

**Review by goranalkovic-infinum — APPROVED** [review]

**Review by piqusy — APPROVED** [review]

**Pull request merged**